src/java.base/share/classes/java/lang/foreign/MemoryLayout.java

```diff
@@ -611,15 +609,17 @@ static PathElement sequenceElement() {
     String toString();
 
     /**
-     * Creates a padding layout with the given size.
+     * Creates a padding layout with the given bitSize and a bit-alignment of eight.
      *
-     * @param size the padding size in bits.
+     * @param bitSize the padding size in bits.
      * @return the new selector layout.
-     * @throws IllegalArgumentException if {@code size <= 0}.
+     * @throws IllegalArgumentException if {@code bitSize < 0} or {@code bitSize % 8 != 0}
      */
-    static PaddingLayout paddingLayout(long size) {
-        MemoryLayoutUtil.checkSize(size);
-        return PaddingLayoutImpl.of(size);
+    static PaddingLayout paddingLayout(long bitSize) {
+        if (bitSize < 0 || bitSize % 8 != 0) {
```

Comment:
If I'm not mistaken, this is the same check that should be applied when we call `withBitAlignment`, right? It might be useful to consolidate (both impl-wise and doc-wise).